src/installer/corehost/cli/hostpolicy/hostpolicy_context.cpp

```diff
@@ -125,6 +125,26 @@ int hostpolicy_context_t::initialize(hostpolicy_init_t &hostpolicy_init, const a
         ++fx_curr;
     }
 
+    pal::string_t fx_resolution;
+    bool is_app = true;
+    for (const auto& fx : fx_definitions)
```

Comment:
This can be folded into the existing loop which wills `app_context_deps_str` right above this as that already iterates over all FX definitions. There will need to be some changes around the "app" framework, but in general it should be done in one place as it's basically the same thing.